pxr/imaging/hdx/pickTask.cpp

```diff
@@ -1158,6 +1186,44 @@ HdxPrimOriginInfo::FromPickHit(HdRenderIndex * const renderIndex,
     return result;
 }
 
+HdxPrimOriginInfo
+HdxPrimOriginInfo::FromPickHit(HdRenderIndex * const renderIndex,
+                               const HdxPickHit &hit)
+{
+    HdSceneIndexBaseRefPtr const sceneIndex =
+        renderIndex->GetTerminalSceneIndex();
+
+    // Fallback value.
+    if (!sceneIndex) {
+        return HdxPrimOriginInfo();
+    }
+
+    return _FromPickHitWithCache(sceneIndex, hit, nullptr);
+}
+
+std::vector<HdxPrimOriginInfo>
+HdxPrimOriginInfo::FromPickHits(HdRenderIndex * const renderIndex,
+                                const std::vector<HdxPickHit> &hits)
+{
+    std::vector<HdxPrimOriginInfo> resultVec;
+    std::map<SdfPath, _InstanceInfo> infoCache;
+
+    resultVec.resize(hits.size());
+    HdSceneIndexBaseRefPtr const sceneIndex =
+        renderIndex->GetTerminalSceneIndex();
+
+    // Fallback value.
+    if (!sceneIndex) {
+        return resultVec;
+    }
+
+    for (int i = 0, n = hits.size(); i < n; i++) {
```

Comment:
Is there a risk that `i` could overflow here because `hits.size()` is a `size_t`?  Could you use `std::transform` here?

Reply:
Yes, I think changing int to size_t is the corerct hing here, thanks.

std::transform I had to look up :) It seems it was introduced in C++17? So my objections to this change would be that it makes USD require C++17 (maybe it does already?).

But more importantly I think such a change would obfuscate the operation of the code (I, for example, would have had to look up std::transform to know what the modified code was doing). If there is some performance advantage to std::transform I'm happy to do it. But otherwise I'd rather leave the old-school code, as purely an aesthetic/readability choice.